liburing/build.rs

```diff
@@ -0,0 +1,45 @@
+extern crate bindgen;
+
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+fn main() {
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    Command::new("cp")
+        .arg("-r")
+        .arg("lib")
+        .arg(out_dir.clone())
+        .status()
+        .expect("copy liburing to out_dir");
+    Command::new("make")
+        .arg("liburing.a")
```

Comment:
~~I think we can't statically link liburing because of LGPL, maybe you can consider using [linux-io-uring](https://github.com/quininer/linux-io-uring).~~

> a) Accompany the work with the complete corresponding machine-readable source code for the Library including whatever changes were used in the work (which must be distributed under Sections 1 and 2 above); and, if the work is an executable linked with the Library, with the complete machine-readable "work that uses the Library", as object code and/or source code, so that the user can modify the Library and then relink to produce a modified executable containing the modified Library. (It is understood that the user who changes the contents of definitions files in the Library will not necessarily be able to recompile the application to use the modified definitions.)

I re-read the LGPL, it not infect LGPL to other libraries, but still has restrictions on final executable.

Reply:
I don't think it's worth to replace original liburing with pure Rust implementation. liburing is kinda standard way and supported library for working with io_uring. work is still in progress there. there is no sense to support another library porting code in par.

Reply:
The developer of `rust-io-uring` has answered a similar [question](https://github.com/stbuehler/rust-io-uring/issues/1), and my point of view is similar to it. I think Rust is better suited to implement this interface than C, and can provide a safer and simpler API.

But now I think using `liburing` is good for `sled`, because `liburing` has a `libaio` [compatibility layer](https://github.com/axboe/liburing/tree/libaio-wrapper). This will provide `sled` with the old version of linux compatibility for free.

Reply:
Thank you for the link. Generally, I think pure Rust implementations are a good idea. In this certain case, it makes no difference, besides that the interface is more natural. I've read mentioned Issue #1, and all that author has to say is his subjective perception of liburing.